orchestrator/crates/orchestrator/src/compression/blob.rs

```diff
@@ -263,3 +265,142 @@ fn da_word(
         JobError::Other(OtherError(color_eyre::eyre::eyre!("Failed to convert decimal string to FieldElement: {}", e)))
     })
 }
+
+pub fn convert_to_biguint(elements: &[Felt]) -> Vec<BigUint> {
+    let input_len = elements.len();
+    if input_len == 0 {
+        return Vec::new(); // Return an empty vector for an empty input
+    }
+
+    // Calculate the required output size: ceil(input_len / 4096.0) * 4096
+    // Integer division trick: (input_len + 4095) / 4096 gives the ceiling division result
+    let num_blocks = (input_len + BLOB_LEN - 1) / BLOB_LEN;
+    let output_len = num_blocks * BLOB_LEN;
+
+    // Initialize the vector with the calculated size, filled with zeros
+    let mut biguint_vec = vec![BigUint::zero(); output_len];
+
+    // Iterate over the input elements and place them in the output vector
+    for (i, element) in elements.iter().enumerate() {
+        // Remove .take(4096)
+        // Convert Felt to [u8; 32]
+        let bytes: [u8; 32] = element.to_bytes_be();
+
+        // Convert [u8; 32] to BigUint
+        let biguint = BigUint::from_bytes_be(&bytes);
+
+        // Place the converted value at the correct index
+        // This automatically leaves remaining spots as zeros
+        biguint_vec[i] = biguint;
+    }
+
+    biguint_vec
+}
```

Comment:
Suggestion: You can simplify this loop by using `.map().chain().collect()` to avoid `resize()`. It  unnecessary allocations.

```suggestion
pub fn convert_to_biguint(elements: &[Felt]) -> Vec<BigUint> {
    let num_blocks = (elements.len() + BLOB_LEN - 1) / BLOB_LEN;
    let output_len = num_blocks * BLOB_LEN;
    let pad_len = output_len - elements.len();

    elements
        .iter()
        .map(|e| BigUint::from_bytes_be(&e.to_bytes_be()))
        .chain(std::iter::repeat(BigUint::zero()).take(pad_len))
        .collect()
}
```

Reply:
Cool! Thanks for the suggestion. implemented it